src/datanode/src/sql/delete.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::ops::Deref;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use common_query::Output;
+use common_telemetry::warn;
+use datatypes::prelude::VectorRef;
+use datatypes::vectors::{BooleanVector, Float64Vector, StringVector, TimestampMillisecondVector};
+use session::context::QueryContextRef;
+use snafu::ResultExt;
+use sql::ast::{BinaryOperator, Expr, Value};
+use sql::statements::delete::Delete;
+use table::engine::TableReference;
+use table::requests::DeleteRequest;
+
+use crate::error::{DeleteSnafu, InvalidSqlSnafu, Result};
+use crate::instance::sql::table_idents_to_full_name;
+use crate::sql::{SqlHandler, SqlRequest};
+
+impl SqlHandler {
+    pub(crate) async fn delete(&self, req: DeleteRequest) -> Result<Output> {
+        let table_ref = TableReference {
+            catalog: &req.catalog_name.to_string(),
+            schema: &req.schema_name.to_string(),
+            table: &req.table_name.to_string(),
+        };
+
+        let table = self.get_table(&table_ref)?;
+
+        let affected_rows = table.delete(req).await.with_context(|_| DeleteSnafu {
+            table_name: table_ref.to_string(),
+        })?;
+
+        Ok(Output::AffectedRows(affected_rows))
+    }
+
+    pub(crate) fn delete_to_request(
+        &self,
+        stmt: Delete,
+        query_ctx: QueryContextRef,
+    ) -> Result<SqlRequest> {
+        let (catalog_name, schema_name, table_name) =
+            table_idents_to_full_name(stmt.table_name(), query_ctx.clone())?;
+        let key_column_values = parser_selection(stmt.selection())?;
+        Ok(SqlRequest::Delete(DeleteRequest {
+            key_column_values,
+            catalog_name,
+            schema_name,
+            table_name,
+        }))
+    }
+}
+
+/// parser selection, currently supported format is `tagkey1 = 'tagvalue1' and tagkey2 = 'tagvalue2'`.
```

Comment:
```suggestion
/// parse selection, currently supported format is `tagkey1 = 'tagvalue1' and tagkey2 = 'tagvalue2'`.
```

I guess you mean `parse`, right?

Reply:
Yes.